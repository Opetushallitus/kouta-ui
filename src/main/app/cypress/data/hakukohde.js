export default () => ({
  oid: '1.2.246.562.20.00000000000000000030',
  toteutusOid: '1.2.246.562.17.00000000000000000001',
  hakuOid: '1.2.246.562.29.00000000000000000001',
  tila: 'tallennettu',
  nimi: { fi: 'Hakukohteen nimi' },
  alkamiskausiKoodiUri: 'kausi_0#1',
  alkamisvuosi: '2024',
  hakulomake: {},
  minAloituspaikat: 100,
  maxAloituspaikat: 150,
  minEnsikertalaisenAloituspaikat: 39,
  maxEnsikertalaisenAloituspaikat: 49,
  kaytetaanHaunAlkamiskautta: false,
  pohjakoulutusvaatimusKoodiUrit: ['pohjakoulutusvaatimustoinenaste_0#1'],
  muuPohjakoulutusvaatimus: {},
  toinenAsteOnkoKaksoistutkinto: true,
  kaytetaanHaunAikataulua: false,
  liitteetOnkoSamaToimitusaika: false,
  liitteetOnkoSamaToimitusosoite: false,
  liitteidenToimitusosoite: null,
  liitteidenToimitustapa: null,
  liitteet: [
    {
      id: '927ac923-a839-4354-9db4-f95d276d9902',
      tyyppiKoodiUri: 'liitetyypitamm_0#1',
      nimi: { fi: 'Nimi' },
      kuvaus: { fi: 'Kuvaus' },
      toimitusaika: '2011-11-11T10:30',
      toimitusosoite: {
        osoite: {
          osoite: { fi: 'Osoite' },
          postinumeroKoodiUri: 'posti_0#2',
        },
        sahkoposti: 'sahkoposti@email.com',
      },
      toimitustapa: 'osoite',
    },
  ],
  valintakokeet: [
    {
      id: 'b9d53560-a7f0-45d3-bd9d-46e67e6049ba',
      tyyppiKoodiUri: 'valintakokeentyyppi_0#1',
      tilaisuudet: [
        {
          osoite: {
            osoite: { fi: 'Osoite' },
            postinumeroKoodiUri: 'posti_0#2',
          },
          aika: { alkaa: '2011-11-11T10:30', paattyy: '2011-11-12T11:45' },
          lisatietoja: { fi: 'Lis√§tietoja' },
        },
      ],
    },
  ],
  hakuajat: [{ alkaa: '2011-11-11T10:30', paattyy: '2011-11-12T11:45' }],
  muokkaaja: '1.2.246.562.24.62301161440',
  organisaatioOid: '1.2.246.562.10.594252633210',
  kielivalinta: ['fi', 'sv'],
  modified: '2019-04-04T08:28',
  kaytetaanHaunHakulomaketta: false,
  hakulomaketyyppi: 'ataru',
  hakulomakeAtaruId: '12345',
  hakulomakeKuvaus: {},
  hakulomakeLinkki: {},
  valintaperusteId: 'b9d53560-a7f0-45d3-bd9d-46e67e6049ba',
});
