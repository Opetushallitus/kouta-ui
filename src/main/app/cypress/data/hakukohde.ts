const hakukohde = () => ({
  oid: '1.2.246.562.20.00000000000000000030',
  toteutusOid: '1.2.246.562.17.00000000000000000001',
  hakuOid: '1.2.246.562.29.00000000000000000001',
  tila: 'tallennettu',
  nimi: { fi: 'Hakukohteen nimi' },
  alkamiskausiKoodiUri: 'kausi_k#1',
  alkamisvuosi: '2024',
  hakulomake: {},
  aloituspaikat: 150,
  ensikertalaisenAloituspaikat: 49,
  kaytetaanHaunAlkamiskautta: false,
  pohjakoulutusvaatimusKoodiUrit: ['pohjakoulutusvaatimuskouta_0#1'],
  pohjakoulutusvaatimusTarkenne: {
    fi: '<p>Fi tarkenne</p>',
  },
  muuPohjakoulutusvaatimus: {},
  toinenAsteOnkoKaksoistutkinto: true,
  kaytetaanHaunAikataulua: false,
  liitteetOnkoSamaToimitusaika: false,
  liitteetOnkoSamaToimitusosoite: false,
  liitteidenToimitusosoite: null,
  liitteidenToimitustapa: null,
  liitteet: [
    {
      id: '927ac923-a839-4354-9db4-f95d276d9902',
      tyyppiKoodiUri: 'liitetyypitamm_4#1',
      nimi: { fi: 'Nimi' },
      kuvaus: { fi: 'Kuvaus' },
      toimitusaika: '2011-11-11T10:30',
      toimitusosoite: {
        osoite: {
          osoite: { fi: 'Osoite' },
          postinumeroKoodiUri: 'posti_00350#2',
        },
        sahkoposti: 'sahkoposti@email.com',
      },
      toimitustapa: 'osoite',
    },
  ],
  metadata: {
    valintakokeidenYleiskuvaus: {
      fi: '<p>Valintakokeiden kuvaus - fi</p>',
    },
  },
  valintakokeet: [
    {
      nimi: {
        fi: 'Kokeen nimi - fi',
      },
      metadata: {
        tietoja: {
          fi: '<p>Tietoa kokeesta - fi</p>',
        },
        liittyyEnnakkovalmistautumista: true,
        ohjeetEnnakkovalmistautumiseen: {
          fi: '<p>Ohjeet ennakkovalmistautumiseen - fi</p>',
        },
        erityisjarjestelytMahdollisia: true,
        ohjeetErityisjarjestelyihin: {
          fi: '<p>Ohjeet erityisj√§rjestelyihin - fi</p>',
        },
      },
      tyyppiKoodiUri: 'tyyppi_1#1',
      tilaisuudet: [
        {
          osoite: {
            osoite: { fi: 'fi osoite' },
            postinumeroKoodiUri: 'posti_00350#1',
          },
          aika: {
            alkaa: '2019-04-16T08:44',
            paattyy: '2019-04-18T08:44',
          },
          lisatietoja: {
            fi: 'fi lisatietoja',
          },
          jarjestamispaikka: {
            fi: 'Jarjestamispaikka - fi',
          },
        },
      ],
    },
  ],
  hakuajat: [{ alkaa: '2011-11-11T10:30', paattyy: '2011-11-12T11:45' }],
  muokkaaja: '1.2.246.562.24.62301161440',
  organisaatioOid: '1.2.246.562.10.594252633210',
  kielivalinta: ['fi', 'sv'],
  modified: '2019-04-04T08:28',
  kaytetaanHaunHakulomaketta: false,
  hakulomaketyyppi: 'ataru',
  hakulomakeAtaruId: '12345',
  hakulomakeKuvaus: {},
  hakulomakeLinkki: {},
  valintaperusteId: 'b9d53560-a7f0-45d3-bd9d-46e67e6049ba',
});

export default hakukohde;
