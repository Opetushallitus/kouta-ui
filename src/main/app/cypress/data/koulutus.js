import { merge } from 'lodash';

const getBaseFields = () => {
  return {
    oid: '1.2.246.562.13.00000000000000000072',
    koulutustyyppi: 'amm',
    koulutusKoodiUri: 'koulutus_0#1',
    tila: 'tallennettu',
    tarjoajat: ['1.2.246.562.10.81934895871'],
    nimi: {
      en: 'Vocational qualification in Agriculture',
      fi: 'Maatalousalan perustutkinto',
      sv: 'Grundexamen inom lantbruksbranschen',
    },
    metadata: {
      tyyppi: 'amm',
      kuvaus: {},
      lisatiedot: [
        {
          otsikkoKoodiUri: 'koulutuksenlisatiedot_0#1',
          teksti: {
            fi: 'koulutuksenlisatiedot_0 kuvaus',
          },
        },
      ],
    },
    julkinen: true,
    muokkaaja: '1.2.246.562.24.62301161440',
    organisaatioOid: '1.2.246.562.10.594252633210',
    kielivalinta: ['fi', 'sv'],
    modified: '2019-04-01T13:01',
  };
};

const getAmmatillinenFields = ({ tyyppi }) => {
  return merge(getBaseFields(), {
    koulutustyyppi: tyyppi,
    koulutusKoodiUri: 'koulutus_0#1',
    tila: 'tallennettu',
    nimi: {
      en: 'Vocational qualification in Agriculture',
      fi: 'Maatalousalan perustutkinto',
      sv: 'Grundexamen inom lantbruksbranschen',
    },
    metadata: { tyyppi },
  });
};

const getKorkeakouluFields = ({ tyyppi }) => {
  return merge(getBaseFields(), {
    johtaaTutkintoon: true,
    koulutustyyppi: tyyppi,
    koulutusKoodiUri: 'koulutus_0#1',
    nimi: { fi: 'Fi nimi', sv: 'Sv nimi' },
    metadata: {
      tyyppi,
      kuvaus: { fi: 'Fi kuvaus', sv: 'Sv kuvaus' },
      tutkintonimikeKoodiUrit: ['tutkintonimikekk_1#1', 'tutkintonimikekk_2#1'],
      opintojenLaajuusKoodiUri: 'opintojenlaajuus_1#1',
      kuvauksenNimi: { fi: 'Fi kuvauksen nimi', sv: 'Sv kuvauksen nimi' },
      koulutusalaKoodiUrit: [
        'kansallinenkoulutusluokitus2016koulutusalataso2_1#1',
      ],
    },
  });
};

const getLukioFields = ({ tyyppi }) => getAmmatillinenFields({ tyyppi });

export default ({ tyyppi = 'amm' } = {}) => {
  if (tyyppi === 'amm') {
    return getAmmatillinenFields({ tyyppi });
  } else if (['yo', 'amk'].includes(tyyppi)) {
    return getKorkeakouluFields({ tyyppi });
  } else if (tyyppi === 'lk') {
    return getLukioFields({ tyyppi });
  }

  return getBaseFields();
};
