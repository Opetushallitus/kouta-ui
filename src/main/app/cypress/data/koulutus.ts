import { merge } from 'lodash/fp';

const getBaseFields = ({ tyyppi = 'amm' } = {}) => {
  return {
    oid: '1.2.246.562.13.00000000000000000072',
    koulutustyyppi: tyyppi,
    koulutuksetKoodiUri: ['koulutus_0#1'],
    tila: 'tallennettu',
    tarjoajat: ['1.2.246.562.10.81934895871'],
    nimi: {
      en: 'Vocational qualification in Agriculture',
      fi: 'Maatalousalan perustutkinto',
      sv: 'Grundexamen inom lantbruksbranschen',
    },
    metadata: {
      tyyppi: tyyppi,
      kuvaus: {},
      lisatiedot: [
        {
          otsikkoKoodiUri: 'koulutuksenlisatiedot_0#1',
          teksti: {
            fi: 'koulutuksenlisatiedot_0 kuvaus',
          },
        },
      ],
    },
    julkinen: true,
    muokkaaja: '1.2.246.562.24.62301161440',
    organisaatioOid: '1.2.246.562.10.594252633210',
    kielivalinta: ['fi', 'sv'],
    modified: '2019-04-01T13:01',
  };
};

const getAmmatillinenFields = ({ tyyppi = 'amm' }) => {
  return merge(getBaseFields(), {
    oid: '1.2.246.562.13.00000000000000000599',
    johtaaTutkintoon: false,
    koulutustyyppi: tyyppi,
    koulutuksetKoodiUri: ['koulutus_351107#12'],
    tila: 'tallennettu',
    tarjoajat: [],
    nimi: {
      en: 'Vocational qualification in Mining',
      fi: 'Kaivosalan perustutkinto',
      sv: 'Grundexamen inom gruvbranschen',
    },
    metadata: {
      tyyppi: tyyppi,
      kuvaus: {},
      lisatiedot: [],
    },
    julkinen: true,
    esikatselu: true,
    muokkaaja: '1.2.246.562.24.26603962037',
    organisaatioOid: '1.2.246.562.10.60198812368',
    kielivalinta: ['fi', 'sv'],
    ePerusteId: 6777660,
    modified: '2020-09-21T16:27',
  });
};

const getKorkeakouluFields = ({ tyyppi }) => {
  return merge(getBaseFields(), {
    johtaaTutkintoon: true,
    koulutustyyppi: tyyppi,
    koulutuksetKoodiUri: ['koulutus_671112#12'],
    nimi: { fi: 'Fysioterapeutti (AMK)', sv: 'Fysioterapeut (YH)' },
    metadata: {
      tyyppi,
      kuvaus: { fi: 'Fi kuvaus', sv: 'Sv kuvaus' },
      tutkintonimikeKoodiUrit: ['tutkintonimikekk_1#1', 'tutkintonimikekk_2#1'],
      opintojenLaajuusKoodiUri: 'opintojenlaajuus_1#1',
      kuvauksenNimi: { fi: 'Fi kuvauksen nimi', sv: 'Sv kuvauksen nimi' },
      koulutusalaKoodiUrit: [
        'kansallinenkoulutusluokitus2016koulutusalataso2_052#1',
      ],
    },
  });
};

const getLukioFields = ({ tyyppi }) => {
  return merge(getBaseFields({ tyyppi }), {
    koulutuksetKoodiUri: ['koulutus_309902#7'],
    nimi: { fi: 'Lukion oppimäärä' },
    metadata: {
      opintojenLaajuusKoodiUri: 'opintojenlaajuus_150',
    },
  });
};

const getTuvaFields = ({ tyyppi }) => {
  return merge(getBaseFields({ tyyppi }), {
    nimi: { fi: 'Tutkintokoulutukseen valmentava koulutus (TUVA)' },
    metadata: {
      opintojenLaajuusKoodiUri: 'opintojenlaajuus_38',
      kuvaus: { fi: 'kuvausteksti' },
    },
  });
};

const getTelmaFields = ({ tyyppi }) => {
  return merge(getBaseFields({ tyyppi }), {
    nimi: { fi: 'Työhön ja itsenäiseen elämään valmentava koulutus (TELMA)' },
    metadata: {
      opintojenLaajuusKoodiUri: 'opintojenlaajuus_60',
      kuvaus: { fi: 'kuvausteksti' },
    },
  });
};

export default ({ tyyppi = 'amm' } = {}) => {
  if (tyyppi.startsWith('amm')) {
    return getAmmatillinenFields({ tyyppi });
  } else if (['yo', 'amk'].includes(tyyppi)) {
    return getKorkeakouluFields({ tyyppi });
  } else if (tyyppi === 'lk') {
    return getLukioFields({ tyyppi });
  } else if (tyyppi === 'tuva') {
    return getTuvaFields({ tyyppi });
  } else if (tyyppi === 'telma') {
    return getTelmaFields({ tyyppi });
  }

  return getBaseFields();
};
