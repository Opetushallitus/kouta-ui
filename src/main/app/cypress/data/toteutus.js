import merge from 'lodash/merge';

const getBaseFields = () => ({
  oid: '1.2.246.562.17.00000000000000000026',
  koulutusOid: '1.2.246.562.13.00000000000000000041',
  tila: 'tallennettu',
  tarjoajat: ['1.2.246.562.10.74534804344'],
  nimi: { fi: 'Koulutuskeskus Salpaus, jatkuva haku, eläintenhoitaja' },
  metadata: {
    kuvaus: {
      fi: 'Toteutuksen kuvaus',
    },
    opetus: {
      koulutuksenAlkamispaivamaara: '2019-08-23T00:00',
      koulutuksenPaattymispaivamaara: '2019-08-26T00:00',
      opetuskieliKoodiUrit: ['oppilaitoksenopetuskieli_1#1'],
      opetuskieletKuvaus: {
        fi: 'Opetuskieli kuvaus',
      },
      opetusaikaKoodiUrit: ['opetusaikakk_1#1'],
      opetusaikaKuvaus: {
        fi: 'Opetusaika kuvaus',
      },
      opetustapaKoodiUrit: ['opetuspaikkakk_1#1'],
      opetustapaKuvaus: {
        fi: 'Opetustapa kuvaus',
      },
      onkoMaksullinen: true,
      maksullisuusKuvaus: {
        fi: 'Maksullisuus kuvaus',
      },
      maksunMaara: 20,
      alkamiskausiKoodiUri: 'kausi_1#1',
      alkamisvuosi: '2024',
      alkamisaikaKuvaus: {
        fi: 'Alkamisaika kuvaus',
      },
      onkoLukuvuosimaksua: true,
      lukuvuosimaksu: 30,
      lukuvuosimaksuKuvaus: {
        fi: 'Lukuvuosimaksu kuvaus',
      },
      onkoStipendia: true,
      stipendinMaara: {
        fi: '90',
      },
      stipendinKuvaus: {
        fi: 'Stipendin kuvaus',
      },
      lisatiedot: [
        {
          otsikkoKoodiUri: 'koulutuksenlisatiedot_0#1',
          teksti: {
            fi: 'koulutuksenlisatiedot_0 kuvaus',
          },
        },
      ],
    },
    asiasanat: [
      { kieli: 'fi', arvo: 'koira' },
      { kieli: 'fi', arvo: 'eläintenhoito' },
    ],
    ammattinimikkeet: [{ kieli: 'fi', arvo: 'eläintenhoitaja' }],
    yhteyshenkilot: [
      {
        nimi: {
          fi: 'Fi nimi',
          sv: 'Sv nimi',
        },
        puhelinnumero: {
          fi: 'Fi puhelinnumero',
          sv: 'Sv puhelinnumero',
        },
        sahkoposti: {
          fi: 'Fi sähköposti',
          sv: 'Sv sähköposti',
        },
        titteli: {
          fi: 'Fi titteli',
          sv: 'Sv titteli',
        },
        wwwSivu: {
          fi: 'Fi verkkosivu',
          sv: 'Sv verkkosivu',
        },
      },
    ],
  },
  lukionLinjanTarkenneKoodiUrit: ['lukiolinjat2_1#1'],
  erityinenKoulutustehtavaKoodiUrit: ['lukionerityistehtava_1#1'],
  muokkaaja: '1.2.246.562.24.62301161440',
  organisaatioOid: '1.2.246.562.10.594252633210',
  kielivalinta: ['fi', 'sv'],
  modified: '2019-03-26T10:19',
});

const getAmmatillinenFields = ({ tyyppi }) =>
  merge(getBaseFields(), {
    metadata: {
      tyyppi,
      osaamisalat: [
        {
          koodi: 'osaamisala_0',
          linkki: {
            fi: 'https://www.salpaus.fi/koulutusesittely/elaintenhoitaja/',
          },
          otsikko: {
            fi: 'Otsikko',
          },
        },
      ],
    },
  });

const getKorkeakouluFields = ({ tyyppi }) =>
  merge(getBaseFields(), {
    metadata: {
      tyyppi,
      ylemmanKorkeakoulututkinnonOsaamisalat: [
        {
          kuvaus: { fi: 'osaamisalan kuvaus' },
          nimi: { fi: 'osaamisalan nimi' },
          linkki: { fi: 'linkki' },
          otsikko: { fi: 'osaamisalan otsikko' },
        },
      ],
      alemmanKorkeakoulututkinnonOsaamisalat: [
        {
          kuvaus: { fi: 'osaamisalan kuvaus' },
          nimi: { fi: 'osaamisalan nimi' },
          linkki: { fi: 'linkki' },
          otsikko: { fi: 'osaamisalan otsikko' },
        },
      ],
    },
  });

const getLukioFields = ({ tyyppi }) =>
  merge(getBaseFields(), {
    metadata: {
      tyyppi,
      opetus: {
        diplomiKoodiUrit: ['lukiodiplomit_1#1'],
        diplomiKuvaus: {
          fi: 'Fi diplomi',
          sv: 'Sv diplomi',
        },
        A1JaA2Kielivalikoima: ['kieli_1#1'],
        B2Kielivalikoima: ['kieli_2#1'],
        B1Kielivalikoima: ['kieli_3#1'],
        B3Kielivalikoima: ['kieli_4#1'],
        aidinkieliKielivalikoima: ['kieli_4#1'],
        muuKielivalikoima: ['kieli_4#1'],
      },
      lukiolinjaKoodiUri: 'lukiolinjat_0#1',
      jaksonKuvaus: {
        fi: 'Fi jakso',
        sv: 'Sv jakso',
      },
    },
  });

export default ({ tyyppi = 'amm' }) => {
  if (tyyppi === 'amm') {
    return getAmmatillinenFields({ tyyppi });
  } else if (['yo', 'amk'].includes(tyyppi)) {
    return getKorkeakouluFields({ tyyppi });
  } else if (tyyppi === 'lk') {
    return getLukioFields({ tyyppi });
  }

  return getBaseFields();
};
