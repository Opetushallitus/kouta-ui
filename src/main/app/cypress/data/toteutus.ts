import _ from 'lodash';

import { Alkamiskausityyppi, ApurahaYksikko } from '#/src/constants';
import { MaksullisuusTyyppi } from '#/src/types/toteutusTypes';

const getBaseFields = () => ({
  oid: '1.2.246.562.17.00000000000000000026',
  koulutusOid: '1.2.246.562.13.00000000000000000041',
  tila: 'tallennettu',
  esikatselu: false,
  tarjoajat: ['1.2.246.562.10.74534804344'],
  nimi: { fi: 'Koulutuskeskus Salpaus, jatkuva haku, eläintenhoitaja' },
  metadata: {
    kuvaus: {},
    opetus: {
      opetuskieliKoodiUrit: ['oppilaitoksenopetuskieli_1#1'],
      opetuskieletKuvaus: {
        fi: 'Opetuskieli kuvaus',
      },
      suunniteltuKestoKuvaus: {
        fi: 'Fi suunniteltuKestoKuvaus',
        sv: 'Sv suunniteltuKestoKuvaus',
      },
      suunniteltuKestoVuodet: 2,
      suunniteltuKestoKuukaudet: 6,
      opetusaikaKoodiUrit: ['opetusaikakk_1#1'],
      opetusaikaKuvaus: {
        fi: 'Opetusaika kuvaus',
      },
      opetustapaKoodiUrit: ['opetuspaikkakk_1#1'],
      opetustapaKuvaus: {
        fi: 'Opetustapa kuvaus',
      },
      maksullisuustyyppi: MaksullisuusTyyppi.MAKSULLINEN,
      maksullisuusKuvaus: {
        fi: 'Maksullisuus kuvaus',
      },
      maksunMaara: 20,
      koulutuksenAlkamiskausi: {
        alkamiskausityyppi: Alkamiskausityyppi.TARKKA_ALKAMISAJANKOHTA,
        koulutuksenAlkamispaivamaara: '2021-04-16T00:00',
        koulutuksenPaattymispaivamaara: '2021-12-12T00:00',
      },
      onkoApuraha: true,
      apuraha: {
        kuvaus: {
          fi: 'Stipendin kuvaus',
        },
        min: 100,
        max: 200,
        yksikko: ApurahaYksikko.EURO,
      },
      lisatiedot: [
        {
          otsikkoKoodiUri: 'koulutuksenlisatiedot_0#1',
          teksti: {
            fi: 'koulutuksenlisatiedot_0 kuvaus',
          },
        },
      ],
    },
    asiasanat: [
      { kieli: 'fi', arvo: 'koira' },
      { kieli: 'fi', arvo: 'eläintenhoito' },
    ],
    ammattinimikkeet: [{ kieli: 'fi', arvo: 'eläintenhoitaja' }],
    yhteyshenkilot: [
      {
        nimi: {
          fi: 'Fi nimi',
          sv: 'Sv nimi',
        },
        puhelinnumero: {
          fi: 'Fi puhelinnumero',
          sv: 'Sv puhelinnumero',
        },
        sahkoposti: {
          fi: 'Fi sähköposti',
          sv: 'Sv sähköposti',
        },
        titteli: {
          fi: 'Fi titteli',
          sv: 'Sv titteli',
        },
        wwwSivu: {
          fi: 'http://fi-verkkosivu',
          sv: 'http://sv-verkkosivu',
        },
      },
    ],
  },
  muokkaaja: '1.2.246.562.24.62301161440',
  organisaatioOid: '1.2.246.562.10.594252633210',
  kielivalinta: ['fi', 'sv'],
  modified: '2019-03-26T10:19',
});

const getAmmatillinenFields = ({ tyyppi }) =>
  _.merge(getBaseFields(), {
    metadata: {
      tyyppi,
      osaamisalat: [
        {
          koodiUri: 'osaamisala_0',
          linkki: {
            fi: 'https://www.salpaus.fi/koulutusesittely/elaintenhoitaja/',
          },
          otsikko: {
            fi: 'Otsikko',
          },
        },
      ],
    },
  });

const getKorkeakouluFields = ({ tyyppi }) =>
  _.merge(getBaseFields(), {
    metadata: {
      tyyppi,
      ylemmanKorkeakoulututkinnonOsaamisalat: [
        {
          kuvaus: { fi: 'osaamisalan kuvaus' },
          nimi: { fi: 'osaamisalan nimi' },
          linkki: { fi: 'http://linkki.com' },
          otsikko: { fi: 'osaamisalan otsikko' },
        },
      ],
      alemmanKorkeakoulututkinnonOsaamisalat: [
        {
          kuvaus: { fi: 'osaamisalan kuvaus' },
          nimi: { fi: 'osaamisalan nimi' },
          linkki: { fi: 'http://linkki.com' },
          otsikko: { fi: 'osaamisalan otsikko' },
        },
      ],
    },
  });

const getLukioFields = ({ tyyppi }) =>
  _.merge(getBaseFields(), {
    metadata: {
      tyyppi,
      yleislinja: true,
      opetus: {
        diplomiKoodiUrit: ['lukiodiplomit_1#1'],
        diplomiKuvaus: {
          fi: 'Fi diplomi',
          sv: 'Sv diplomi',
        },
      },
      kielivalikoima: {
        A1JaA2Kielet: ['kieli_1#1'],
        B2Kielet: ['kieli_2#1'],
        B1Kielet: ['kieli_3#1'],
        B3Kielet: ['kieli_4#1'],
        aidinkielet: ['kieli_4#1'],
        muutKielet: ['kieli_4#1'],
      },
      jaksonKuvaus: {
        fi: 'Fi jakso',
        sv: 'Sv jakso',
      },
    },
  });

export default ({ tyyppi = 'amm' }) => {
  if (tyyppi.startsWith('amm')) {
    return getAmmatillinenFields({ tyyppi });
  } else if (['yo', 'amk'].includes(tyyppi)) {
    return getKorkeakouluFields({ tyyppi });
  } else if (tyyppi === 'lk') {
    return getLukioFields({ tyyppi });
  }

  return getBaseFields();
};
