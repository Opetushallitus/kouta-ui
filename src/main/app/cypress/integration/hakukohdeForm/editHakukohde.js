import merge from 'lodash/merge';

import { getByTestId, chooseKieliversiotLanguages } from '../../utils';

import koulutus from '../../data/koulutus';
import toteutus from '../../data/toteutus';
import valintaperuste from '../../data/valintaperuste';
import hakukohde from '../../data/hakukohde';
import { stubHakukohdeFormRoutes } from '../../hakukohdeFormUtils';

const fillKieliversiotSection = cy => {
  getByTestId('kieliversiotSection', cy).within(() => {
    chooseKieliversiotLanguages(['fi'], cy);
  });
};

const tallenna = cy => {
  getByTestId('tallennaHakukohdeButton', cy).click({ force: true });
};

describe('createHakukohdeForm', () => {
  const organisaatioOid = '1.1.1.1.1.1';
  const toteutusOid = '2.1.1.1.1.1';
  const koulutusOid = '3.1.1.1.1';
  const hakuOid = '4.1.1.1.1.1';
  const valintaperusteOid = '5.1.1.1.1.1';
  const hakukohdeOid = '6.1.1.1.1.1';

  const testHakukohdeFields = {
    toteutusOid,
    hakuOid,
    organisaatioOid,
    oid: hakukohdeOid,
  };

  beforeEach(() => {
    cy.server();

    stubHakukohdeFormRoutes({ cy, organisaatioOid, hakuOid });

    cy.route({
      method: 'GET',
      url: `**/toteutus/${toteutusOid}`,
      response: merge(toteutus({ tyyppi: 'amm' }), {
        oid: toteutusOid,
        organisaatioOid: organisaatioOid,
        koulutusOid: koulutusOid,
      }),
    });

    cy.route({
      method: 'GET',
      url: `**/koulutus/${koulutusOid}`,
      response: merge(koulutus({ tyyppi: 'amm' }), {
        oid: koulutusOid,
        organisaatioOid: organisaatioOid,
      }),
    });

    cy.route({
      method: 'GET',
      url: '**/valintaperuste/list**',
      response: [
        merge(valintaperuste(), {
          oid: valintaperusteOid,
          nimi: { fi: 'Valintaperusteen nimi' },
        }),
      ],
    });

    cy.visit(`/hakukohde/${hakukohdeOid}/muokkaus`);
  });

  it('should be able to edit hakukohde', () => {
    cy.route({
      method: 'GET',
      url: `**/hakukohde/${hakukohdeOid}`,
      response: merge(hakukohde(), testHakukohdeFields),
    });

    cy.route({
      method: 'POST',
      url: '**/hakukohde',
      response: {
        muokattu: false,
      },
    }).as('updateHakukohdeRequest');

    fillKieliversiotSection(cy);
    tallenna(cy);

    cy.wait('@updateHakukohdeRequest').then(({ request }) => {
      cy.wrap(request.body).snapshot();
    });
  });
});
