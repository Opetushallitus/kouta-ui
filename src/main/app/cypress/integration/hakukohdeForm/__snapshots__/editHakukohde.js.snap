exports[`editHakukohdeForm > should be able to edit hakukohde #0`] =
{
  "alkamiskausiKoodiUri": "kausi_0#1",
  "alkamisvuosi": 2024,
  "aloituspaikat": 150,
  "ensikertalaisenAloituspaikat": 49,
  "hakuOid": "4.1.1.1.1.1",
  "hakuajat": [
    {
      "alkaa": "2011-11-11T10:30",
      "paattyy": "2011-11-12T11:45"
    }
  ],
  "hakulomake": {},
  "hakulomakeAtaruId": "12345",
  "hakulomakeKuvaus": {},
  "hakulomakeLinkki": {},
  "hakulomaketyyppi": "ataru",
  "kaytetaanHaunAikataulua": false,
  "kaytetaanHaunAlkamiskautta": false,
  "kaytetaanHaunHakulomaketta": false,
  "kielivalinta": [
    "fi"
  ],
  "liitteet": [
    {
      "kuvaus": {
        "fi": "Kuvaus"
      },
      "nimi": {
        "fi": "Nimi"
      },
      "toimitusaika": "2011-11-11T10:30",
      "toimitusosoite": {
        "osoite": {
          "osoite": {
            "fi": "Osoite"
          },
          "postinumeroKoodiUri": "posti_0#2"
        },
        "sahkoposti": "sahkoposti@email.com"
      },
      "toimitustapa": "osoite",
      "tyyppiKoodiUri": "liitetyypitamm_0#1"
    }
  ],
  "liitteetOnkoSamaToimitusaika": false,
  "liitteetOnkoSamaToimitusosoite": false,
  "liitteidenToimitusaika": null,
  "liitteidenToimitusosoite": null,
  "liitteidenToimitustapa": null,
  "modified": "2019-04-04T08:28",
  "muokkaaja": "1.2.246.562.24.62301161440",
  "muuPohjakoulutusvaatimus": {},
  "nimi": {
    "fi": "Hakukohteen nimi"
  },
  "oid": "6.1.1.1.1.1",
  "organisaatioOid": "1.1.1.1.1.1",
  "pohjakoulutusvaatimusKoodiUrit": [
    "pohjakoulutusvaatimustoinenaste_0#1"
  ],
  "pohjakoulutusvaatimusTarkenne": {
    "fi": "<p>Fi tarkenne</p>"
  },
  "tila": "tallennettu",
  "toinenAsteOnkoKaksoistutkinto": true,
  "toteutusOid": "2.1.1.1.1.1",
  "valintakokeet": [
    {
      "tilaisuudet": [
        {
          "aika": {
            "alkaa": "2011-11-11T10:30",
            "paattyy": "2011-11-12T11:45"
          },
          "lisatietoja": {
            "fi": "Lis√§tietoja"
          },
          "osoite": {
            "osoite": {
              "fi": "Osoite"
            },
            "postinumeroKoodiUri": "posti_0#2"
          }
        }
      ],
      "tyyppiKoodiUri": "valintakokeentyyppi_0#1"
    }
  ],
  "valintaperusteId": "b9d53560-a7f0-45d3-bd9d-46e67e6049ba"
};
